apache-http-client/src/main/java/com/joyent/http/signature/apache/httpclient/HttpSignatureAuthScheme.java

```diff
@@ -202,18 +204,10 @@ protected Header signRequestHeader(final Credentials credentials,
             request.setHeader(HttpHeaders.DATE, stringDate);
         }
 
-        final String authz;
-
-        try {
-            authz = signer.get().createAuthorizationHeader(
-                    login, fingerprint, keyPair, stringDate);
-        } catch (HttpSignatureException e) {
-            String details = String.format("Unable to authenticate [%s] with "
-                    + "fingerprint [%s] using keypair [%s]",
-                    login, fingerprint, keyPair);
-            throw new AuthenticationException(details, e);
-        }
+        signatureCacheMap.computeIfAbsent(credentials, NEW_CACHE_FUNCTION);
```

Comment:
I was thinking of something where the jitter at the top of the second would be problematic, but a bulk object store probably isn't the right tool for something so low latency anyway.